contracts/contracts/PoolDeposits.sol

```diff
@@ -0,0 +1,155 @@
+import './NoLossDao.sol';
+import '@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/ERC20.sol';
+
+contract PoolDeposits is Initializable {
+  using SafeMath for uint256;
+
+  address public admin;
+  uint256 public totalDepositedDai;
+  mapping(address => uint256) public depositedDai;
+
+  uint256 public proposalAmount; // Add mapping of the proposal amount per iteration...
+
+  ///////// DEFI Contrcats ///////////
+  IERC20 public daiContract;
+  IAaveLendingPool public aaveLendingContract;
+  IADai public adaiContract;
+  INoLossDao public noLossDaoContract;
+  address public aaveLendingContractCore;
+
+  // Mods
+
+  modifier noLossDaoContractOnly() {
+    require(
+      address(noLossDaoContract) == msg.sender, // Is this a valid way of getting the address?
+      'function can only be called by no Loss Dao contract'
+    );
+    _;
+  }
+
+  function initialize(
+    address daiAddress,
+    address aDaiAddress,
+    address aavePoolAddress,
+    address aavePoolCoreAddress,
+    address noLossDaoAddress
+  ) public initializer {
+    daiContract = IERC20(daiAddress);
+    aaveLendingContract = IAaveLendingPool(aavePoolAddress);
+    adaiContract = IADai(aDaiAddress);
+    aaveLendingContractCore = aavePoolCoreAddress;
+    noLossDaoContract = INoLossDao(noLossDaoAddress);
+    admin = msg.sender;
+  }
+
+  /**
+    * @dev Returns the DAI deposited by this user
+    * @param userAddress the user you would like to check
+    * ?? I guess we don't need this function we could simply call the value from the contract
+    */
+  function usersDeposit(address userAddress) external view returns (uint256) {
+    return depositedDai[userAddress];
+  }
+
+  /**
+    * @dev Lets a user join DAOcare through depositing
+    * @param amount the user wants to deposit into the DAOcare pool
+    */
+  function deposit(uint256 amount) public {
+    // This is so much less granular now which I don't like
+    require(
+      noLossDaoContract.canDeposit(msg.sender) == true,
+      'User not eligible to deposit'
+    );
+
+    // Get from aave lending pool the latest address....
+    daiContract.transferFrom(msg.sender, address(this), amount);
+    daiContract.approve(address(aaveLendingContractCore), amount);
+    aaveLendingContract.deposit(address(daiContract), amount, 0);
+
+    // Set here
+    depositedDai[msg.sender] = amount;
+    totalDepositedDai = totalDepositedDai.add(amount);
+
+    // Set in NoLossDAo
+    noLossDaoContract.setUserIterationJoined(msg.sender);
+  }
+
+  /**
+     * @dev Lets a user withdraw their original amount sent to DAOcare
+     */
+  function withdrawDeposit() public {
+    require(
+      noLossDaoContract.canWithdraw(msg.sender) == true,
+      'User not eligible to withdraw'
+    );
+
+    uint256 amount = depositedDai[msg.sender];
+
+    depositedDai[msg.sender] = 0;
+    totalDepositedDai = totalDepositedDai.sub(amount);
+
+    noLossDaoContract.resetUserIterationJoined(msg.sender);
```

Comment:
Just an asside, given we will probably combine these function calls. But when ever you run logic on another contract it could be better to check that the result of that logic was successful (lets say it returns a boolean).

Reply:
I see. And then revert in the case it is doesn't return true I guess? Won't it fail and revert anyway if the call to the other contract fails

Reply:
Yes (as long as we don't put a kind of try-catch around it), if the other contract reverts everything will revert.
It is just a design decision, if we want to let this contract check if it was successful or not, or just blindly trust that the other contract revents when it should. It is the difference between `send` and `transfer` really. I prefer it returning a bool (it can also revert internally) because this contract (the caller) can then create its own revent and revert message.

But its very much a personal taste thing, I prefer honey to sugar you ? :package: